File: Aula 01/tecweb/src/main/java/br/com/davidbuzatto/tecweb/mensagem/Forma.java
package br.com.davidbuzatto.tecweb.mensagem;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * Uma forma geométrica genérica.
 * 
 * @author Prof. Dr. David Buzatto
 */
@Data
@AllArgsConstructor
public class Forma {
    
    private String tipo;
    
    @JsonProperty( "xIni" )
    private int xIni;
    
    @JsonProperty( "yIni" )
    private int yIni;
    
    @JsonProperty( "xFim" )
    private int xFim;
    
    @JsonProperty( "yFim" )
    private int yFim;
    
    private String corContorno;
    private String corPreenchimento;
    
}

File: Aula 01/tecweb/src/main/resources/static/js/forma.js
export class Forma {
    
    xIni;
    yIni;
    xFim;
    yFim;
    
    corContorno;
    corPreenchimento;
    
    larguraContorno = 3;
    
    constructor( xIni, yIni, xFim, yFim, corContorno, corPreenchimento ) {
        this.xIni = xIni;
        this.yIni = yIni;
        this.xFim = xFim;
        this.yFim = yFim;
        this.corContorno = corContorno;
        this.corPreenchimento = corPreenchimento;
    }
    
    desenhar( ctx ) {
    }
    
}